From ca3cb92bbea2e83bb5459602c63aaf0717119d4a Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Thu, 29 Jul 2021 18:21:08 -0700
Subject: [PATCH 20/24] bsg: Fix build error with CONFIG_BLK_DEV_BSG_COMMON=m

When CONFIG_BLK_DEV_BSG_COMMON is enabled as a module, which can happen
when CONFIG_SCSI=m and CONFIG_BLK_DEV_BSGLIB=n, the following error
occurs:

In file included from arch/x86/kernel/asm-offsets.c:13:
In file included from include/linux/suspend.h:5:
In file included from include/linux/swap.h:9:
In file included from include/linux/memcontrol.h:22:
In file included from include/linux/writeback.h:14:
In file included from include/linux/blk-cgroup.h:23:
include/linux/blkdev.h:539:26: error: field has incomplete type 'struct
bsg_class_device'
        struct bsg_class_device bsg_dev;
                                ^
include/linux/blkdev.h:539:9: note: forward declaration of 'struct
bsg_class_device'
        struct bsg_class_device bsg_dev;
               ^
1 error generated.

The definition of struct bsg_class_device is kept under an #ifdef
directive, which does not work when CONFIG_BLK_DEV_BSG_COMMON is a
module, as the define is CONFIG_BLK_DEV_BSG_COMMON_MODULE.

Use IS_ENABLED instead, which evaluates to 1 when
CONFIG_BLK_DEV_BSG_COMMON is y or m.

Fixes: 78011042684d ("scsi: bsg: Move bsg_scsi_ops to drivers/scsi/")
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20210730012108.3385990-1-nathan@kernel.org
---
 include/linux/bsg.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/bsg.h b/include/linux/bsg.h
index b887da20bd41..9602ae3ab01b 100644
--- a/include/linux/bsg.h
+++ b/include/linux/bsg.h
@@ -7,7 +7,7 @@
 struct request;
 struct request_queue;
 
-#ifdef CONFIG_BLK_DEV_BSG_COMMON
+#if IS_ENABLED(CONFIG_BLK_DEV_BSG_COMMON)
 struct bsg_ops {
 	int	(*check_proto)(struct sg_io_v4 *hdr);
 	int	(*fill_hdr)(struct request *rq, struct sg_io_v4 *hdr,
-- 
2.32.0.264.g75ae10bc75

